src/background/tab-state.js

```diff
@@ -183,14 +183,22 @@ export default function TabState(initialState, onchange) {
       pendingAnnotationCountRequests.set(tabId, {
         cancel: () => {
           clearTimeout(timerId);
-          resolve(0);
+          reject('Badge request cancelled');
```

Comment:
As a convention we try to ensure that promises are always rejected with `Error` instances (or `Error`-like objects). This is convenient when catching the error as you can make some assumptions about the type of the exception value (eg. that it always has a `message` property).

We have a set of existing error subclasses defined in `errors.js`, so I would recommend extending that.

Reply:
I agree.

I added a new error.